File: Python/A_Project/B_Snake_and_Food/Utility/score_service.py
class ScoreService:
    def __init__(self):
        self.score = 0

    def add(self, item):
        if item is not None:
            self.score += item.get_points()

    def get_score(self) -> int:
        return self.score

    def reset(self):
        self.score = 0

File: Ruby/A_Project/B_Snake_And_Food/Utility/score_service.rb
# frozen_string_literal: true

class ScoreService
  def initialize
    @score = 0
  end

  def add(item)
    @score += item.get_points if !item.nil?
  end

  def get_score
    @score
  end

  def reset
    @score = 0
  end
end
